lib/usecases/run/EndRunUseCase.js

```diff
@@ -50,8 +61,32 @@ class EndRunUseCase {
             const runEntity = await RunRepository.findOne(queryBuilder);
 
             if (runEntity) {
-                runEntity.timeO2End = timeO2End;
-                runEntity.timeTrgEnd = timeTrgEnd;
+                if (runQuality) {
+                    runEntity.runQuality = runQuality;
+                }
+                if (timeO2End && timeO2End > 0) {
+                    runEntity.timeO2End = timeO2End;
+                }
+                if (timeO2Start && timeO2Start > 0) {
+                    runEntity.timeO2Start = timeO2Start;
+                }
+                if (timeTrgEnd && timeTrgEnd > 0) {
+                    runEntity.timeTrgEnd = timeTrgEnd;
+                }
+                if (timeTrgStart && timeTrgStart > 0) {
+                    runEntity.timeTrgStart = timeTrgStart;
+                }
+                runEntity.trgGlobalRunEnabled = trgGlobalRunEnabled;
+                runEntity.trgEnabled = trgEnabled;
+                if (pdpConfigOption) {
+                    runEntity.pdpConfigOption = pdpConfigOption;
+                }
+                if (pdpTopologyDescriptionLibraryFile) {
+                    runEntity.pdpTopologyDescriptionLibraryFile = pdpTopologyDescriptionLibraryFile;
+                }
+                if (tfbDdMode) {
+                    runEntity.tfbDdMode = tfbDdMode;
+                }
```

Comment:
Maybe we can do it more straightforward it, using `spread syntax` or extract a function, because we will probably do the same many times (patch existing object) :
```js
runEntity = {
    ...runEntity,
    ...body.runQuality || runEntity.runQuality
    // etc
}
for the fields that do not need supplementary checks ?
```

Reply:
Opted for using `runEntity.update(body)`